.github/workflows/functions-deploy.yml

```diff
@@ -0,0 +1,90 @@
+# TODO - currently this is only called from the `internal` deploy job. It should be part of all envs
+name: Deploy Node.js project to Azure Function App
+
+on:
+  workflow_call:
+    inputs:
+      ENVIRONMENT:
+        required: true
+        type: string
+      REGISTRY:
+        required: true
+        type: string
+      REPO:
+        required: true
+        type: string
+      APP:
+        required: true
+        type: string
+    secrets:
+      AZURE_CLIENT_ID:
+        required: true
+      AZURE_TENANT_ID:
+        required: true
+      AZURE_SUBSCRIPTION_ID:
+        required: true
+      AZURE_FUNCTIONAPP_PUBLISH_PROFILE:
+        required: true
+
+permissions:
+  contents: read
+
+env:
+  # AZURE_FUNCTIONAPP_NAME has to match the azurerm_linux_function_app's name in functions.tf
+  AZURE_FUNCTIONAPP_NAME: 'polling-function-${{ inputs.ENVIRONMENT }}'
+  AZURE_FUNCTIONAPP_PACKAGE_PATH: 'azure_functions'
+  NODE_VERSION: '20.x'
+
+jobs:
+  build-and-deploy:
```

Comment:
We may need a comment explaining why the build and deploy steps are combined for this workflow.